libraries/botbuilder-dialogs/src/waterfallDialog.ts

```diff
@@ -108,20 +109,23 @@ export class WaterfallDialog<O extends object = {}> extends Dialog<O> {
     private async runStep(dc: DialogContext, index: number, reason: DialogReason, result?: any): Promise<DialogTurnResult> {
         if (index < this.steps.length) {
             // Update persisted step index
-            const state = dc.activeDialog.state as WaterfallDialogState;
+            const state: WaterfallDialogState = dc.activeDialog.state as WaterfallDialogState;
             state.stepIndex = index;
 
             // Create step context
-            let nextCalled = false;
+            const nextCalled: boolean = false;
             const step: WaterfallStepContext<O> = {
                 index: index,
                 options: <O>state.options,
                 reason: reason,
                 result: result,
                 values: state.values,
-                next: async (result?: any) => {
-                    if (nextCalled) { throw new Error(`WaterfallStepContext.next(): method already called for dialog and step '${this.id}[${index}]'.`) }
-                    return await this.dialogResume(dc, DialogReason.nextCalled, result);
+                next: async (result2?: DialogTurnResult<any>): Promise<DialogTurnResult<any>> => {
```

Comment:
I don't know if this is correct, and as someone who prefers more descriptive variable and argument names, `result2` isn't terribly helpful.

I think the more correct code here is this:

```js
next: async (stepResult?: any): Promise<DialogTurnResult<any>> => {
```
Otherwise TypeScript devs will have to return a pass in a `DialogTurnResult` which is in turn wrapped in another `DialogTurnResult`.

Reply:
Thanks for busting me on that lazy variable name!